Vpc flow without lambda function

sonofagl1tch opened the pull request:
Closes #6


modified template to write vpc flow logs directly to s3 rather than through a lambda function


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

sonofagl1tch committed:
added apache user to wheel group so bad things can happen

sonofagl1tch committed:
modified template to write vpc flow logs directly to s3 rather than through a lambda function

mgmacias95 reviewed (approved):
Hello @sonofagl1tch,

VPC Flow logs are written to the S3 bucket correctly but I'm not getting alerts in Wazuh. I think the reason could be that the path specified in the configuration is different than the path created in the S3 bucket:

![imagen](https://user-images.githubusercontent.com/8604906/46371851-c3a52780-c689-11e8-9b42-26a4d525f64e.png)
```xml
  <bucket type="custom">
   <name>detonationlab-s3bucketvpcflow-2ok03g4b33e4</name>
<path>flowlogs</path>
   <access_key>xxxxxx</access_key>
   <secret_key>xxxxx</secret_key>
  </bucket>
```
In my configuration I have `flowlogs`, in my bucket I have `vpcflowlogs`. 

I'm approving this PR, since VPC Flow logs are written in S3 buckets (this is what this PR does), but I'd like this to be reviewed and fixed in a separate PR.

Best regards,
Marta

mgmacias95 commented:
Hi @sonofagl1tch,

I found out why Wazuh is not able to read logs using this. I have opened an issue in the wazuh repository (https://github.com/wazuh/wazuh/issues/1516). **This PR should NOT be merged until the Wazuh team fixes this issue**.

Best regards,
Marta

Commit:
Merge branch 'master' into VPCFlow-without-lambda-function

Commit:
Merge branch 'master' into VPCFlow-without-lambda-function

mgmacias95 commented:
Hello @sonofagl1tch,

I was trying to test this again and I wasn't able to deploy it. I get the following error:
![imagen](https://user-images.githubusercontent.com/8604906/50048837-06e23480-00d6-11e9-91ae-510b355a049b.png)
It seems it's necessary to remove the lambda function previously used to retrieve VPC logs.

Best regards,
Marta

sonofagl1tch committed:
removed lambda functions

sonofagl1tch commented:
> Hello @sonofagl1tch,
> 
> I was trying to test this again and I wasn't able to deploy it. I get the following error:
> ![imagen](https://user-images.githubusercontent.com/8604906/50048837-06e23480-00d6-11e9-91ae-510b355a049b.png)
> It seems it's necessary to remove the lambda function previously used to retrieve VPC logs.
> 
> Best regards,
> Marta

found that the lambda resources were still in the tempate. once i removed them the template loaded successfully.

mgmacias95 commented on awsDetonationLab.template:
```diff
@@ -876,107 +876,27 @@
-        "Handler": "index.handler",
-        "MemorySize": 256,
-        "Role": {
+        "DeliverLogsPermissionArn": {
```
According to the [AWS::EC2::FlowLog documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-flowlog.html#w2ab1c21c10d102c40b9):
> If you specify `LogDestinationType` as `s3`, do not specify `DeliverLogsPermissionArn` or `LogGroupName`. 

Which means this field is no longer necessary and must be removed.

sonofagl1tch replied:
removed vpc lambda related resources from template

mgmacias95 replied:
Aren't those vpc lambda related too?
https://github.com/sonofagl1tch/AWSDetonationLab/blob/a681a7465d78f038f6aba1ee54a4cdcbfe185d10/awsDetonationLab.template#L847-L851
https://github.com/sonofagl1tch/AWSDetonationLab/blob/a681a7465d78f038f6aba1ee54a4cdcbfe185d10/awsDetonationLab.template#L879-L917
https://github.com/sonofagl1tch/AWSDetonationLab/blob/a681a7465d78f038f6aba1ee54a4cdcbfe185d10/awsDetonationLab.template#L867-L878

sonofagl1tch committed:
removed vpc lambda related resources from template

sonofagl1tch commented:
@mgmacias95 this is ready for review again

Commit:
Remove extra vpc lambda related resources from template

Commit:
Update Wazuh AWS configuration to work with new VPC flow mode

Commit:
Fix spacing in cloudtrail configuration

mgmacias95 reviewed (approved):
Hello @sonofagl1tch,

I think the code is ready in our side, but further testing needs to be done when Wazuh releases the new AWS integration including support for this PR.

Best regards,
Marta

Commit:
Upgrade Wazuh installation to 3.8.0-6.5.4

mgmacias95 commented:
Hi there,

I have tested this PR using Wazuh v3.8.0. The following additional permissions need to be added to the user the detonation lab creates:
```javascript
       {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "ec2:DescribeFlowLogs",
            "Resource": "*"
        },
```

Best regards,
Marta

Commit:
Add DescribeFlowLogs policy

mgmacias95 commented:
A new policy with the necessary permissions has been added in commit https://github.com/sonofagl1tch/AWSDetonationLab/pull/21/commits/cc2149a9b7e8854428af0ec686958ad7ff5351a0. 

The PR works as expected except for issue https://github.com/sonofagl1tch/AWSDetonationLab/issues/61 which I think must be fixed in a separate PR.

@sonofagl1tch this is ready to merge! 😄

sonofagl1tch commented:
The new Iam policy you created is over privileged.  Using an asterisk for resources means that any user/role/group that uses that policy can run those commands against any reside in your account. I suggest looking at how I created previous access policies in the template so we can restrict access to only detonation lab related resources. Let's talk offline about this before the commit.

mgmacias95 commented:
Hello @sonofagl1tch,

I tried modifying the policy to make it resource specific as you suggested and got the following warning:
![imagen](https://user-images.githubusercontent.com/8604906/51633664-790a2c80-1f52-11e9-991b-7d7a2b6ab336.png)
Which roughly translates as _"the action needs access to all resources_".

Best regards,
Marta

Commit:
Remove enable auth command

authd server is enabled by default in wazuh v3.8. These lines are no longer necessary.

Commit:
Upgrade wazuh from v3.8.0 to v3.8.1

Commit:
Bump version to Wazuh 3.8.2 and Elastic 6.6.0

sonofagl1tch commented:
after researching through VPC documentation, the new feature to all VPC flow to write directly to S3 does not support an option to restrict the ec2:DescribeFlowLogs api call to specific flows. a feature request to the vpc flow team will need to be opened to correct this over priveleged feature. This PR should be good to approve and merge.

Commit:
Merge branch 'master' into VPCFlow-without-lambda-function
